src/test/java/com/imsweb/layout/LayoutUtilsTest.java

```diff
@@ -269,4 +271,92 @@ public void testReadWriteHl7Layout() throws IOException {
         }
     }
 
+    @Test
+    public void testReadLayout() throws IOException {
+        //Run similar tests as above, reading different layouts using generic method
+
+        //Test with comma separated layout
+        CommaSeparatedLayoutXmlDto commaSeparatedLayout = new CommaSeparatedLayoutXmlDto();
```

Comment:
This is true for setting the fields on the layouts and also asserting them. The purpose of this method is to handle different type of layouts transparently. You know that it's actually using the same code than the individual read methods. So there is no need to exhaustively re-test every single field of the layout. Try to create your testing layout with just the minimum information (just the required fields). And when you assert them, just assert the ID, I don't think it's useful to assert anything else; that stuff is already unit tested in the individual read methods.